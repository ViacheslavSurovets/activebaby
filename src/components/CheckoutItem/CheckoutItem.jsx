import React from 'react';
import { connect } from 'react-redux';
import {
  CheckoutItemContainer,
  ImageContainer,
  QuantityContainer,
  RemoveButtonContainer,
  TextContainer
} from './styles';
import { clearItemFromCart, addItem, removeItem } from '@redux/cart/cart.actions';
import PropTypes from 'prop-types';

const CheckoutItem = ( { cartItem, clearItem, addItem, removeItem } ) => {
  const { title, images, price, quantity } = cartItem;

  return (
    <CheckoutItemContainer>
      <ImageContainer>
        <img src={ images[ 0 ] } alt='item' />
      </ImageContainer>
      <TextContainer>{ title }</TextContainer>
      <QuantityContainer>
        <div onClick={ () => removeItem ( cartItem ) }>&#10094;</div>
        <span>{ quantity }</span>
        <div onClick={ () => addItem ( cartItem ) }>&#10095;</div>
      </QuantityContainer>
      <TextContainer>{ price }</TextContainer>
      <RemoveButtonContainer onClick={ () => clearItem ( cartItem ) }>
        &#10005;
      </RemoveButtonContainer>
    </CheckoutItemContainer>
  );
};

const mapDispatchToProps = dispatch => ({
  clearItem: item => dispatch ( clearItemFromCart ( item ) ),
  addItem: item => dispatch ( addItem ( item ) ),
  removeItem: item => dispatch ( removeItem ( item ) )
});

export default connect ( null, mapDispatchToProps ) ( CheckoutItem );


CheckoutItem.propTypes = {
  cartItem: PropTypes.object,
  clearItem: PropTypes.func,
  addItem: PropTypes.func,
  removeItem: PropTypes.func
};

