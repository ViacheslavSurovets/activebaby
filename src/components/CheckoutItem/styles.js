import styled from 'styled-components';

export const CheckoutItemContainer = styled.div`
  width: 100%;
  display: flex;
  min-height: 10rem;
  border-bottom: 1px solid ${ ( { theme } ) => theme.darkGray };
  padding: 1.5rem 0;
  font-size: 2rem;
  align-items: center;
  user-select: none;
`;

export const ImageContainer = styled.div`
  width: 23%;
  padding-right: 1.5rem;

  img {
    width: 100%;
    height: 100%;
  }
`;

export const TextContainer = styled.span`
  width: 23%;
`;

export const QuantityContainer = styled ( TextContainer )`
  display: flex;

  span {
    margin: 0 1rem;
  }

  div {
    cursor: pointer;
  }
`;

export const RemoveButtonContainer = styled.div`
  padding-left: 1.2rem;
  cursor: pointer;
`;

