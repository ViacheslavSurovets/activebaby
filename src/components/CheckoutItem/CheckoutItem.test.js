import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import CheckoutItem from './CheckoutItem';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('CheckoutItem', ()=> {
  it('CheckoutItem Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <CheckoutItem  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
