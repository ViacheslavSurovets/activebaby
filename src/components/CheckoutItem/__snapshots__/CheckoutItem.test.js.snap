// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem CheckoutItem Snapshot 1`] = `ShallowWrapper {}`;
